lib/utils/wrap.js

```diff
@@ -22,3 +22,7 @@ export const wrap = (window['ShadyDOM'] && window['ShadyDOM']['noPatch'] && wind
   window['ShadyDOM']['wrap'] :
   (window['ShadyDOM'] ? (n) => ShadyDOM['patch'](n) : (n) => n);
 
+// Wrapping a local operation (e.g. appendChild) is only needed iff noPatch is
+// explicitly true, and on-demand patching is not used.
+export const wrapIfNeeded = (window['ShadyDOM'] && window['ShadyDOM']['noPatch'] === true && window['ShadyDOM']['wrap']) ?
```

Comment:
Let's add comment to list known cases where `wrap` and not `wrapIfNeeded` is required:
* When moving (`appendChild`/`insertBefore`) a node without an explicit `removeChild` to an unpatched location: in this case it needs to be unlinked from logical tree, have its className unscoped, trigger `slotchange` and, trigger distribution if it was a `<slot>`
* When setting `className` to an unpatched node: if it was in a SR it needs to go through ShadyCSS to avoid losing its scoping class
* When calling `querySelector` on an unpatched node: it won't return scoped results
* When getting `textContent` or `innerHTML` on an unpatched node: it won't return scoped results

Also, now wondering if `wrap` and `wrapAlways` might be better naming (and less annoying to boot)?

Reply:
Added.